src/components/Snippet/index.tsx

```diff
@@ -0,0 +1,240 @@
+  border-radius: ${({ theme }) => theme.radii.default};
+`
+
+const StyledStack = styled(Stack)`
+  width: 100%;
+`
+
+const ButtonContainer = styled('div', {
+  shouldForwardProp: prop => !['multiline'].includes(prop),
+})<{ multiline?: boolean }>`
+  position: absolute;
+  top: 0;
+  right: 0;
+  padding: ${({ theme, multiline }) =>
+    multiline
+      ? `${theme.space['2']} ${theme.space['2']} 0 0`
+      : `${theme.space['1']} ${theme.space['2']} 0 0`};
+  background: ${({ theme }) => theme.colors.neutral.backgroundStrong};
+  border-radius: ${({ theme }) => theme.radii.default};
+  border: 2px solid transparent;
+  ${({ multiline, theme }) =>
+    !multiline
+      ? `box-shadow: -27px 0 19px -11px ${theme.colors.neutral.backgroundStrong}`
+      : ''};
+`
+
+const ShowMoreContainer = styled('div', {
+  shouldForwardProp: prop => !['showMore'].includes(prop),
+})<{ showMore?: boolean }>`
+  width: 100%;
+  box-shadow: ${({ theme, showMore }) =>
+    !showMore
+      ? `0px -22px 19px -6px
+    ${theme.colors.neutral.backgroundStrong}`
+      : 'none'};
+`
+
+const StyledButton = styled.button`
+  width: 100%;
+  background: none;
+  border: none;
+  padding: ${({ theme }) => theme.space['2']};
+  padding-top: ${({ theme }) => theme.space['1']};
+  cursor: pointer;
+`
+
+const AlignCenterText = styled(Text)`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`
+
+const AnimatedArrowIcon = styled(Icon, {
+  shouldForwardProp: prop => !['showMore'].includes(prop),
+})<{ showMore?: boolean }>`
+  transform: ${({ showMore }) =>
+    showMore ? 'rotate(180deg)' : 'rotate(0deg)'};
+  transform-origin: center;
+  transition: transform 300ms ease-in-out;
+`
+
+type SnippetProps = {
+  value: string
+  multiline?: boolean
+  /**
+   * prefix display an element at the beginning of the snippet that is not copiable or selectable.
+   * For `lines` prefix it will display the line number.
+   * For `command` prefix it will display a `$` sign.
+   */
+  prefix?: Prefixes
+  showText?: string
+  hideText?: string
+} & Pick<CopyButtonProps, 'copyText' | 'copiedText'>
+
+const Snippet = ({
+  value,
+  copyText,
+  copiedText,
+  multiline = false,
+  showText = 'Show',
+  hideText = 'Hide',
+  prefix,
+}: SnippetProps) => {
+  const theme = useTheme()
+  const [showMore, setShowMore] = useState(false)
+
+  const lines = value.split(LINES_BREAK_REGEX).filter(Boolean)
+  const numberOfLines = lines.length
+  const hasShowMoreButton = numberOfLines > 4 && multiline
+
+  /**
+   * This is used to calculate the max height of the snippet in order to add an animation when the show more button is clicked.
+   * The max height is calculated by multiplying the number of lines by the line height of the text plus the padding (2 times).
+   */
+  const maxHeightExtended =
+    numberOfLines * Number(theme.typography.code.lineHeight.replace('px', '')) +
+    Number(theme.space['2'].replace('px', '')) * 2
```

Comment:
Seems hella complicated, could we instead have a height: fit-content and an animation with a keyframes ?

Reply:
@chambo-e that's the tricky thing about `height` animation. I cannot use `height` as value for animation I have to use `max-height` so I cannot set `height: fit-content`. I have set animation with `height` from `120px` to `fit-content` here is the result:

https://user-images.githubusercontent.com/15812968/199985245-9b939d00-d280-4b0c-aafb-9ba681024f55.mov

I believe that `auto`, `fit-content`, etc values are excluded from transition and animations.
Here is what I've found about it:
<img width="783" alt="Screenshot 2022-11-04 at 14 38 47" src="https://user-images.githubusercontent.com/15812968/199986152-0a49e0b7-711a-4465-a3d5-e0092abd781d.png">
https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Transitions/Using_CSS_transitions#defining_transitions

The best way I found is to calculate the number of lines and so the size of the height and sending it to the CSS to be able to make a smooth animation. 

https://user-images.githubusercontent.com/15812968/199985874-b69f13b7-402e-4093-b32c-0482b7beb518.mov